PWG/EMCAL/EMCALtasks/AliEmcalAodTrackFilterTask.cxx

```diff
@@ -219,16 +209,140 @@ AliEmcalAodTrackFilterTask* AliEmcalAodTrackFilterTask::AddTaskEmcalAodTrackFilt
     mgr->ConnectInput(aodTask, 0,  cinput1 );
   
     return aodTask;
+}}
+void AliEmcalAodTrackFilterTask::ExecOnce(){
+
+  AliParticleContainer* OutputCont= static_cast<AliParticleContainer*>(fParticleCollArray.Last());
+    // add tracks to event if not yet there
+    fTracksOut->Delete();
+    if(OutputCont->GetIsEmbedding()){
+          AliVEvent *event = AliEmcalContainerUtils::GetEvent(InputEvent(),kTRUE);
+          if(!(event->FindListObject(fTracksOutName))){
+          event->AddObject(fTracksOut);
+          OutputCont->SetArray(event);}
+          }
+    else if (!(InputEvent()->FindListObject(fTracksOutName))) {
+      InputEvent()->AddObject(fTracksOut);
+          OutputCont->SetArray(InputEvent());
+    }
```

Comment:
You can simplify this by using `AliVEvent *event = AliEmcalContainerUtils::GetEvent(InputEvent(), OutputCont->GetIsEmbedding())` and then just using that event. However, it's not really critical, so I'm fine with just approving as is